airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialLoadHandler.java

```diff
@@ -155,7 +166,16 @@ private static boolean isCompositePrimaryKey(final ConfiguredAirbyteStream strea
     return stream.getStream().getSourceDefinedPrimaryKey().size() > 1;
   }
 
-  public static long calculateChunkSize() {
-    return DEFAULT_CHUNK_SIZE; // TEMP
+  public static long calculateChunkSize(final TableSizeInfo tableSizeInfo, final AirbyteStreamNameNamespacePair pair) {
+    // If table size info could not be calculated, a default chunk size will be provided.
+    if (tableSizeInfo == null || tableSizeInfo.tableSize() == 0 || tableSizeInfo.avgRowLength() == 0) {
+      LOGGER.info("Chunk size could not be determined for pair: {}, defaulting to {} rows", pair, DEFAULT_CHUNK_SIZE);
+      return DEFAULT_CHUNK_SIZE;
+    }
+    final long avgRowLength = tableSizeInfo.avgRowLength();
+    final long chunkSize = QUERY_TARGET_SIZE_GB / avgRowLength;
```

Comment:
handle case where `avgRowLength` is zero.

Reply:
ln 173 will take care of this case.